src/Models/Behaviors/HasRelated.php

```diff
@@ -51,6 +51,11 @@ public function loadRelated($browser_name)
         return $this->relatedCache[$browser_name] = $this->relatedItems
             ->where('browser_name', $browser_name)
             ->map(function ($item) {
+                if (filled($item->related) && !isset($item->related->position))
+                {
+                    $item->related->position = $item->position ?? null;
+                }
+
```

Comment:
Hey @antonioribeiro,

I had a look at this and I feel it will be better if we treat "related" as a pivot table?

```php
->map(function ($item) {
  /** @var \A17\Twill\Models\Model $model */
  $model = $item->related;

  $model->setRelation('pivot', $item);

  return $model;
```

This way, you can do `Page::getRelated('test')->first()->pivot->position`

What do you think?

Reply:
<img width="839" alt="Screenshot 2022-06-10 at 09 12 40" src="https://user-images.githubusercontent.com/866743/173010775-b56d2a94-c4c3-44e0-b422-5981b67107d0.png">